### PR TITLE
Add on: pull_request: to Push GHA workflow

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
Following up on https://github.com/Codecademy/docs/pull/407, @jakemhiller is helping me with GHA.

### CODE REVIEW

**Comment by JoshuaKGoldberg** [comment]
oy vey.

Screenshot of the Push / ci (compile) workflow happening in both (pull_request) and (push)

**Review by jakemhiller — APPROVED** [review]
makes sense, i would think this could duplicate some CI runs though, since both events will fire at the same time in some cases. that can be fixed if it's an issue thought

File: .github/workflows/push.yml [hunk 1]
```diff
@@ -13,4 +13,6 @@ jobs:
       matrix:
         command: ["compile", "format:verify", "lint:js", "lint:md", "test"]
 
-on: push
+on:
+  push:
+  pull_request:
```
Comment (jakemhiller) [inline .github/workflows/push.yml:18]: this might make more sense?

```suggestion
on:
  push:
    branches:
      - main
  pull_request:
```

**Comment by JoshuaKGoldberg** [comment]
Yeah that makes a ton of sense, thanks Jake!

**Comment by JoshuaKGoldberg** [comment]
Let's try it out 🚀

**Review by Dusch4593 — COMMENTED** [review]
LGTM! Can't wait to try this out 😃

**Pull request merged**